src/lib.rs

```diff
@@ -78,6 +102,23 @@ fn test_NumberOrString() {
     );
 }
 
+
+pub trait UriSource {
+    fn get_uri(&self) -> &Url;
+    
+    fn parse_file_path(&self) -> Result<PathBuf, Box<Error>> {
+        let uri = self.get_uri();
+        
+        if uri.scheme() != "file" {
+            return Err("URI scheme is not `file`".into());
+        }
+        
+        uri.to_file_path().map_err(|_err| "Invalid file path in URI".into())
+    }
+}
```

Comment:
Oddly to_file_path does not check that schema is file. So there is still value in an util function that checks for that. But yeah, it can be a free function, the UrlSource trait is not longer neeeded (and I'll move the util function to rls)